File: bench/python/simd_map.py
# SIMD cross-language benchmark: elementwise (x * 2.0 + 1.0). Matches bench/simd_map.march
import time

n = 5_000_000
xs = [(i % 100) / 100.0 for i in range(n)]
t0 = time.perf_counter()
mapped = [x * 2.0 + 1.0 for x in xs]
t1 = time.perf_counter()
total = 0.0
for x in mapped:
    total += x
print(f"RESULT {total}")
print(f"TIME_MS {(t1 - t0) * 1000.0}")

File: bench/rust/simd_map.rs
// SIMD cross-language benchmark: elementwise (x * 2.0 + 1.0). Matches bench/simd_map.march
use std::time::Instant;

fn main() {
    let n: usize = 5_000_000;
    let arr: Vec<f64> = (0..n).map(|i| (i % 100) as f64 / 100.0).collect();
    let t0 = Instant::now();
    let mapped: Vec<f64> = arr.iter().map(|x| x * 2.0 + 1.0).collect();
    let elapsed = t0.elapsed();
    let total: f64 = mapped.iter().sum();
    println!("RESULT {}", total);
    println!("TIME_MS {}", elapsed.as_secs_f64() * 1000.0);
}
